### ISSUE TITLE
Cross compilation broken on OSX: wrong archiver flags

### ISSUE THREAD

**Issue by flxo** [esrlabs/bake#178]
As already mentioned in #176 it is not possible to overwrite the archiver flags. The patch for fixing #176 statically decides about the archiver flags based on the host. The following scenario is broken:

* Host OSX
* (Cross) Toolchain with a `ar` as archiver

`bake` will pick the `libtool` flags because of the host type. This doesn't work if the archiver command is set to some `ar` variant (common case for cross compilation).

**Comment by aschaal** [comment]
Yes, you are right. I will add an option to overwrite these flags.

**Comment by flxo** [comment]
Thanks @aschaal. Ping me for real world test before a release.

**Comment by aschaal** [comment]
After talking to flxo this fix is not urgently needed, to it will be shifted.
Internal starndard flags will be removed in bake3, which should fix this automatically.
